### DISCUSSION TITLE
Flashing Micropython on custom mimxrt/Teensy 4.x

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#11523] [m.iMX RT / Teensy 4.x] [answered]
Hi,

what is the correct procedure to flash a custom board using the mimxrt chip like the one on the Teensy 4.x? The Teensy has a M0 on board to handle flashing apparently... is it possible to omit this one and flash the mimxrt1602 directly?
Thanks!

**Comment by ned-pcs** [comment, marked as answer]
The i.MXRT parts have a serial loader in ROM that allows for loading over USB, serial, CAN, etc.
The "[NXP MCU Boot Utility](https://github.com/JayHeng/NXP-MCUBootUtility)" lets you load a flash chip over USB or serial.

Alternatively, you could use a JLink or similar to program the flash via a small flash driver program loaded into RAM.
